src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -543,6 +543,21 @@ def resolve_imports_and_packages(
         else session.get_session_stage()
     )
 
+    # resolve packages
```

Comment:
Package resolution should happen before import resolution, because resolving packages can lead to imports. (for example, if local files/folders are defined in requirements.txt)